src/PostgREST/Request/ApiRequest.hs

```diff
@@ -389,6 +389,16 @@ userApiRequest conf@AppConfig{..} dbStructure req reqBody
           o = rangeOffset ro
   ranges = M.insert "limit" (rangeIntersection headerRange (fromMaybe allRange (M.lookup "limit" urlRange))) urlRange
 
+  limitFromQparams :: [(Text, Maybe ByteString)] -> Maybe Integer
+  limitFromQparams [] = Nothing
+  limitFromQparams ((k, v):qs) = if isLimitQueryParameter then readMaybe . toS =<< v else limitFromQparams qs
+    where isLimitQueryParameter = isJust v && endingIn ["limit"] k
+
+  invalidLimit :: Bool
+  invalidLimit = case limitFromQparams qParams of 
+      Just n -> n < 0
```

Comment:
Could you add some tests to ensure `?limit=-1` gives an error? Maybe for embedded resources as well(`?clients.limit=-1`)

Reply:
@steve-chavez 
There is actually a test for negative limit already in `RangeSpec.hs` :
```
it "fails if limit is negative" $
        get "/items?select=id&limit=-1"
          `shouldRespondWith` [json|{"message":"HTTP Range error"}|]
          { matchStatus  = 416
          , matchHeaders = [matchContentTypeJson]
          }
```

I found this while I was trying to make it throw a 400 in case of negative limit and adding a test for it in QuerySpec.hs

I will leave it as 416 now, because I don't know what things downstream are depending on a 416 error when a negative limit is passed. But I'll add another test for embedded resources.